WcaOnRails/app/models/user.rb

```diff
@@ -331,6 +331,18 @@ def avatar_requires_wca_id
     end
   end
 
+  def send_on_create_confirmation_instructions
+    NewRegistrationMailer.send_registration_mail(self).deliver_now
+  end
+
+  def send_reconfirmation_instructions
+    @reconfirmation_required = false
+
+    unless @skip_confirmation_notification
+      ReconfirmationMailer.send_reconfirmation_mail(self).deliver_now
```

Comment:
Actually, I'm concerned that this is going to break reconfirmation. This line of code looks a lot different than the `def send_confirmation_instructions` https://github.com/plataformatec/devise/blob/b8f644198d34e96039b84b69842ecb4edffa4664/lib/devise/models/confirmable.rb#L115-L122 that is currently happening.

Can we simply not override this method, and instead edit WcaOnRails/app/views/devise/mailer/confirmation_instructions.html.erb?

Reply:
the lines of code you are talking about are actually `send_*re*confirmation_instructions`, not `send_confirmation_instructions`. And besides the changed method call, the methods are identical! I did not change anything about `send_confirmation_instructions`, so we're still using Devise's method for this, but `send_reconfirmation_instructions` does not call `send_confirmation_instructions` anymore but the newly created mailer instead.

Reply:
Sorry, my comment wasn't very clear. Let me try again:

The method you've written here looks identical to devise's, **except** for the fact that devises's calls `send_confirmation_instructions`, and yours calls `ReconfirmationMailer.send_reconfirmation_mail(self).deliver_now`. I'm concerned that this is not a safe change, because `send_confirmation_instructions` looks like it does a number of things (such as sometimes generating a confirmation token?), and the new code you've written doesn't do that.

It doesn't feel like this `send_reconfirmation_instructions` method is supposed to be overridden.